src/auro-carousel.js

```diff
@@ -42,14 +58,188 @@ class AuroCarousel extends LitElement {
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            entry.target.removeAttribute('tabindex');
+            entry.target.removeAttribute('aria-hidden');
+          } else {
+            entry.target.setAttribute('tabindex', '-1');
+            entry.target.setAttribute('aria-hidden', true);
+          }
+        })
+      }
+
+      this.intersectionObserver = new IntersectionObserver(callback, options);
+      this.observeChildren();
+    }
+  }
+
+  /**
+   * Internal method to set up the resize observer. When an element is resized, we update
+   * the scroll flags so the buttons are shown/hidden appropriately.
+   * @private
+   * @return {void}
+   */
+  setUpResizeObserver() {
+    if ('ResizeObserver' in window) {
+      // Update scrollflags when a child's size changes (e.g., a custom element becomes
+      // defined or an image is loaded)
+      this.resizeObserver = new ResizeObserver(() => this.setScrollFlags());
+      Array.from(this.children).forEach((element) => {
+        this.resizeObserver.observe(element);
+      });
+    }
+  }
+
+  /**
+   * Internal method to observe the components children with IntersectionObserver.
+   * @private
+   * @return {void}
+   */
+  observeChildren() {
+    Array.from(this.children).forEach((element) => {
+      this.intersectionObserver.observe(element);
+    })
+  }
+
+  /**
+   * Internal method to handle slot change. When the slot contents change, we need to re-observe the
+   * slotted elements.
+   * @private
+   * @return {void}
+   */
+  handleSlotChange() {
+    this.setScrollFlags(false);
+    if (this.intersectionObserver) {
+      this.intersectionObserver.disconnect();
+      this.observeChildren();
+    }
+  }
+
+  /**
+   * Internal method to handle clicks on the shoulder buttons.
+   * @private
+   * @param {boolean} scrollRight - whether to scroll the carousel right or left.
+   * @return {void}
+   */
+  handleClick(scrollRight) {
+    if (scrollRight) {
+      this.scrollCarousel(this.scrollDistance)
+    } else {
+      this.scrollCarousel(-1 * this.scrollDistance);
+    }
+  }
 
-  // function that renders the HTML and CSS into  the scope of the component
   render() {
+    const carouselClassMap = {
+      "carousel": true,
+      "carousel--scrolledToStart": this.scrolledToStart,
+      "carousel--scrolledToEnd": this.scrolledToEnd
+    }
+
     return html`
-      <div class=${this.cssClass}>
-        <slot></slot>
+      ${iconProperties}
+      <div class="breakpoint-detector"></div>
+      <div role="group" 
+        aria-label="${ifDefined(this.label)}" 
+        aria-roledescription="carousel"
+        class="${classMap(carouselClassMap)}" 
+        @scroll=${() => this.setScrollFlags(true)} >
+        <button @click=${() => this.handleClick(false)} class="button button--left">
+          ${unsafeSVG(chevronLeftIcon.svg)}<span class="util_displayHiddenVisually">Scroll left</span>
```

Comment:
Consider using auro-icon

Reply:
I'm having some issues with auro-icon so I'd like to leave that for a later enhancement. The issues are as follows:
- When running with polymer serve, icon import fails with `"Error resolving module specifier “@alaskaairux/icons/dist/icons/terminal/plane-diag-fill_es6.js”. Relative module specifiers must start with “./”, “../” or “/”."` I tried the plane-diag-fill icon since that was the icon used on the docsite.
- When bundling auro-carousel, the polyfills chunk has ballooned to 640 KB and seems to be including every icon.

I'll open these issues in the auro-icon repo, but I don't think it's worth holding up the carousel release to troubleshoot them.

Reply:
Re: the polyfills bundle, you can see the same behavior in the bundled auro-icon: https://unpkg.com/@alaskaairux/auro-icon@1.0.1/dist/polyfills.js

Reply:
Fixed with latest auro-icon version.